src/clj/game/cards/upgrades.clj

```diff
@@ -1681,8 +1681,10 @@
 (defcard "Tucana"
   (let [ability {:async true
                  :prompt "Choose a piece of ice to install and rez"
-                 :choices (req (cancellable (filter ice? (:deck corp))))
-                 :effect (req (corp-install state side eid target nil {:install-state :rezzed :combined-credit-discount 3}))
+                 :choices (req (cancellable (filter ice? (:deck corp)) true))
+                 :effect (req (do (corp-install state side eid target nil {:install-state :rezzed :combined-credit-discount 3})
+                                  (shuffle! state :corp :deck)
+                                  (system-msg state side (str "shuffles R&D"))))
```

Comment:
The way our "async" stuff works is kind of awkward. You have to use the macro `wait-for` and then explicitly "complete" the callback associated with the `eid` at the end:

```clojure
(req (wait-for (corp-install state side (make-eid state eid) target nil {:install-state :rezzed :combined-credit-discount 3})
               (shuffle! state :corp :deck)
               (system-msg state side "shuffles R&D")
               (effect-completed state side eid)))
```

Something like that. Make sure it compiles but it should be good to go otherwise.

Reply:
I ran into some issues with the tests complaining about an already used eid.  I modified this to add `make-eid` for the corp install, not sure if that was correct.

Reply:
Just reread your suggested change and you actually had `make-eid` in there, I just forgot to copy it!